lib/features/onboarding/ui/pages/screens/basic_info_screen.dart

```diff
@@ -1,45 +1,59 @@
 import 'package:flutter/material.dart';
 import 'package:flutter/services.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
 import 'package:miny_design_system/miny_design_system.dart';
 
-import '../../utilities/onboarding_constants.dart';
-import '../widgets/bottom_action_bar.dart';
-import '../widgets/onboarding_title.dart';
+import '../../../store/onboarding_store.dart';
+import '../../../utilities/onboarding_constants.dart';
+import '../../widgets/bottom_action_bar.dart';
+import '../../widgets/miny_divider.dart';
+import '../../widgets/onboarding_title.dart';
 
-class BasicInfoPage extends StatefulWidget {
-  final Function(String name) onTap;
-  final String number;
-  final String email;
+class BasicInfoScreen extends ConsumerStatefulWidget {
+  final VoidCallback onTap;
 
-  const BasicInfoPage({
+  const BasicInfoScreen({
     super.key,
     required this.onTap,
-    required this.number,
-    required this.email,
   });
 
   @override
-  State<BasicInfoPage> createState() => _BasicInfoPageState();
+  ConsumerState<BasicInfoScreen> createState() => _BasicInfoPageState();
 }
 
-class _BasicInfoPageState extends State<BasicInfoPage> {
-  String? _fullNameErrorText;
+class _BasicInfoPageState extends ConsumerState<BasicInfoScreen> {
+  String? _errorText;
+  late UserProfileStore store;
+  late TextEditingController fullNameController;
```

Comment:
change it to nameController also private 
```suggestion
  late TextEditingController _nameController;
```